pootle/apps/pootle_statistics/migrations/0011_cleanup_submissions_squashed_0023_remove_scorelog.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2017-05-11 04:59
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import F
+
+
+CHECK_FIELD_FLAG = 5
+MUTED = 0
+MUTE_FLAG = 6
+SUGG_ACCEPT_FLAG = 3
+STATE_FIELD_FLAG = 3
+UNMUTED = 1
+UNMUTE_FLAG = 7
+WEB_FLAG = 1
+
+CLEANUP_STORE_DATA_SQL = (
+    "UPDATE `pootle_store_data` "
+    " INNER JOIN `pootle_app_submission` "
+    "  ON `pootle_app_submission`.`id` = `pootle_store_data`.`last_submission_id` "
+    " SET `last_submission_id` = NULL"
+    " WHERE (`pootle_app_submission`.`field` = 1"
+    "        OR `pootle_app_submission`.`type` in (10, 2, 8, 9))")
+CLEANUP_TP_DATA_SQL = (
+    "UPDATE `pootle_tp_data` "
+    " INNER JOIN `pootle_app_submission` "
+    "  ON `pootle_app_submission`.`id` = `pootle_tp_data`.`last_submission_id` "
+    " SET `last_submission_id` = NULL"
+    " WHERE (`pootle_app_submission`.`field` = 1"
+    "        OR `pootle_app_submission`.`type` in (10, 2, 8, 9))")
+CLEANUP_SQL = (
+    "DELETE FROM `pootle_app_submission` "
+    " WHERE (`unit_id` IS NULL"
+    "        OR `field` = 1"
+    "        OR `type` in (10, 2, 8, 9))")
+
+
+def cleanup_subs_with_sql(schema_editor):
+    cursor = schema_editor.connection.cursor()
+    cursor.execute(CLEANUP_STORE_DATA_SQL)
+    cursor.execute(CLEANUP_TP_DATA_SQL)
+    cursor.execute(CLEANUP_SQL)
+
+
+def cleanup_subs_with_orm(apps):
+    # type 10 is the previous SubmissionTypes.UNIT
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    (subs.filter(field=1)
+     | subs.filter(type__in=[2, 8, 9, 10])).delete()
+
+
+def cleanup_subs(apps, schema_editor):
+    if schema_editor.connection.vendor == "mysql":
+        cleanup_subs_with_sql(schema_editor)
+    else:
+        cleanup_subs_with_orm(apps)
+
+
+def _assoc_accept_suggestion_subs(apps):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    suggestions = apps.get_model("pootle_store.Suggestion").objects
+    suggestions = suggestions.filter(state__name="accepted").values(
+        "review_time", "pk", "unit_id", "reviewer")
+    subs = subs.filter(
+        type=SUGG_ACCEPT_FLAG).filter(field=STATE_FIELD_FLAG)
+
+    for suggestion in suggestions.iterator():
+        suggestion["submitter"] = suggestion["reviewer"]
+        suggestion["creation_time"] = suggestion["review_time"]
+        suggestion_id = suggestion["pk"]
+        del suggestion["review_time"]
+        del suggestion["reviewer"]
+        del suggestion["pk"]
+        subs.filter(**suggestion).update(suggestion_id=suggestion_id)
+
+
+def _delete_redundant_accept_sugg_subs(apps):
+    subs = apps.get_model("pootle_statistics.Submission").objects
+    subs = subs.filter(type=SUGG_ACCEPT_FLAG)
+    # these appear to be caused by dupes
+    to_delete = (
+        subs.filter(suggestion__isnull=True)
+        | subs.exclude(creation_time=F("suggestion__review_time"))
```

Comment:
`subs.filter(suggestion__isnull=True)` can't be moved above because it depends on first other code populating this field.